deployment/terraform/assets/cluster.tf

```diff
@@ -210,17 +251,17 @@ resource "aws_instance" "proxy_server" {
 
 resource "aws_iam_user" "s3user" {
   name  = "${var.cluster_name}-s3user"
-  count = var.app_instance_count > 1 && var.s3_external_bucket_name == "" ? 1 : 0
+  count = var.app_instance_count > 1 && var.s3_external_bucket_name == "" && !var.create_efs ? 1 : 0
```

Comment:
Ohh boy.

Reply:
Oh boy indeed :joy: Maybe we should make use of modules for this. Having an s3 module that can be automatically switched on/off. Something like https://stackoverflow.com/a/72867545/3248221. I haven't ever tried that, so if that's too difficult/cumbersome, we could at least refactor this whole `var.app_instance_count > 1 && var.s3_external_bucket_name == "" && !var.create_efs ? 1 : 0` into a new `create_s3_resources` variable. Thoughts?